File: Data Structures & Algorithms/binary-tree-diameter/submission-8.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int diameterOfBinaryTree(TreeNode root) {
        int [] result = new int [1];
        
        maxDepth(root, result);
        return result[0];
    }

    public int maxDepth(TreeNode root, int [] result){
        if(root == null){
            return 0;
        }

        int leftResult = maxDepth(root.left, result);
        int rightResult = maxDepth(root.right, result);

        result[0] = Math.max(result[0], leftResult + rightResult);

        return Math.max(leftResult, rightResult) + 1;
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-8.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:

        result = []

        queue = deque()
        queue.append(root)

        while queue:
            max_value = 0

            for i in range(len(queue)):
                current_node = queue.popleft()

                if current_node:
                    max_value = max(max_value, current_node.val)
                    queue.append(current_node.left)
                    queue.append(current_node.right)
            
            if max_value != 0:
                result.append(max_value)
            
        return result


        